nym-node-status-api/src/db/queries/testruns.rs

```diff
@@ -0,0 +1,156 @@
+use crate::{
+    db::models::{TestRunDto, TestRunStatus},
+    testruns::now_utc,
+};
+use futures_util::TryStreamExt;
+use nym_bin_common::models::ns_api::TestrunAssignment;
+use serde::Deserialize;
+use sqlx::{pool::PoolConnection, Sqlite};
+
+pub(crate) async fn get_testruns(conn: PoolConnection<Sqlite>) -> anyhow::Result<Vec<TestRunDto>> {
+    // TODO dz accept mut reference, repeat in all similar functions
+    let mut conn = conn;
+    let testruns = sqlx::query_as!(
+        TestRunDto,
+        r#"SELECT
+            id as "id!",
+            gateway_id as "gateway_id!",
+            status as "status!",
+            timestamp_utc as "timestamp_utc!",
+            ip_address as "ip_address!",
+            log as "log!"
+         FROM testruns
+         WHERE status = 0
+         ORDER BY timestamp_utc"#
+    )
+    .fetch(&mut *conn)
+    .try_collect::<Vec<_>>()
+    .await?;
+
+    Ok(testruns)
+}
+
+pub(crate) async fn get_testrun_by_id(
+    conn: &mut PoolConnection<Sqlite>,
+    testrun_id: i64,
+) -> anyhow::Result<TestRunDto> {
+    sqlx::query_as!(
+        TestRunDto,
+        r#"SELECT
+            id as "id!",
+            gateway_id as "gateway_id!",
+            status as "status!",
+            timestamp_utc as "timestamp_utc!",
+            ip_address as "ip_address!",
+            log as "log!"
+         FROM testruns
+         WHERE id = ?
+         ORDER BY timestamp_utc"#,
+        testrun_id
+    )
+    .fetch_one(&mut *conn)
+    .await
+    .map_err(|err| {
```

Comment:
fyi: `anyhow` has a useful `context()` method that does this very thing

Reply:
do you mean this?
```rust
.map_err(|err| {
    anyhow::anyhow!(
        "Couldn't retrieve testrun {}: {}",
        testrun_id,
        err.context()
    )
})
```

Reply:
you mean like this?
```rust
.fetch_one(&mut *conn)
.await
.context(format!("Couldn't retrieve testrun {testrun_id}"))
```

Reply:
yup